### PR TITLE
Enable type test for @fluidframework/app-insights-logger

### PR DESCRIPTION
**Pull request by chentong7**
## Description

> With the app-insights-logger now marked as @beta in anticipation of the beta release, it's important to implement type-testing within the package. This will help us stay informed of any changes that could potentially break the API at the type level.

### CODE REVIEW

File: examples/client-logger/app-insights-logger/package.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
 	"scripts": {
 		"build": "fluid-build . --task build",
 		"build:compile": "fluid-build . --task compile",
+		"build:test": "tsc --project tsconfig.json",
```
Comment (Josmithr) [inline examples/client-logger/app-insights-logger/package.json:20]: Is this necessary? This seems unrelated to type-tests.

**Review by Josmithr — COMMENTED** [review]
This PR should presumably add the generated type-test code. Have you run the tests and verified that the file is being generated? We'll want to check that in.

File: examples/client-logger/app-insights-logger/package.json [hunk 2]
```diff
@@ -53,6 +56,7 @@
 		"react-dom": "^17.0.1"
 	},
 	"devDependencies": {
```
Comment (Josmithr) [inline examples/client-logger/app-insights-logger/package.json:58]: You'll need to add a dev dependency on the "previous" version of this package. Take a look at one of our other packages using the type tests for an example.

File: examples/client-logger/app-insights-logger/package.json (hunk 1 above)
Comment (chentong7) [reply examples/client-logger/app-insights-logger/package.json:20]: I thought this is the type test... Adding test class here

File: packages/framework/client-logger/app-insights-logger/package.json [hunk 3]
```diff
@@ -91,7 +91,8 @@
 		"test": "npm run test:mocha",
 		"test:mocha": "mocha --recursive \"dist/test/**/*.spec.*js\" -r node_modules/@fluidframework/mocha-test-setup",
 		"test:mocha:verbose": "cross-env FLUID_TEST_VERBOSE=1 npm run test:mocha",
-		"tsc": "tsc"
+		"typetests:gen": "fluid-type-test-generator",
+		"typetests:prepare": "flub typetests --dir . --reset --previous --normalize"
```
Comment (alexvy86) [inline packages/framework/client-logger/app-insights-logger/package.json:95]: I don't think the removal of the `tsc` script was intentional, right?

Comment (chentong7) [reply packages/framework/client-logger/app-insights-logger/package.json:95]: It's being complained by `flub policy check` that `tsc` has no dependency on, so I removed it.

Comment (alexvy86) [reply packages/framework/client-logger/app-insights-logger/package.json:95]: Hmm let's look at the error, removing this script means we would lose one of the two builds (CJS, ESM) that we currently do for this package, so it definitely doesn't seem like the way to go 😄 .

Comment (chentong7) [reply packages/framework/client-logger/app-insights-logger/package.json:95]: You are right. It does result the index.d.ts output file missing. Adding it back but it fails on `flub policy check` error. Looking into it...

Comment (chentong7) [reply packages/framework/client-logger/app-insights-logger/package.json:97]: I think we do need to have the `build:test` script here, since our main entry is JS file. Referring to PR: https://github.com/microsoft/FluidFramework/pull/18939

File: packages/framework/client-logger/app-insights-logger/package.json [hunk 4]
```diff
@@ -75,6 +75,7 @@
 		"build:compile": "fluid-build . --task compile",
 		"build:docs": "fluid-build . --task api",
 		"build:esnext": "tsc-multi --config ../../../../common/build/build-common/tsc-multi.esm.json",
+		"build:test": "tsc --project ./tsconfig.json",
```
Comment (alexvy86) [inline packages/framework/client-logger/app-insights-logger/package.json:78]: What exactly is the error if we remove this script? The package doesn't have a separate tests-only build like others do (with a separate tsconfig file inside src/test, that gets used for this script). So I'm thinking of either removing this script if we can skip that specific policy for this package, or changing it to something like `"build:test": "echo 'No separate tests-only build'"` (which might also require some policy exclusion).

Comment (chentong7) [reply packages/framework/client-logger/app-insights-logger/package.json:78]: It was like:
~~~
client-release-group-root: error during command 'flub check policy' (exit code 1)
client-release-group-root: WARNING:
client-release-group-root: file failed the "fluid-build-tasks-tsc" policy: packages/framework/client-logger/app-insights-logger/package.json
client-release-group-root: 'tsc' task is missing the following dependency:
client-release-group-root:      typetests:gen
client-release-group-root: Statistics: 7844 processed, 107 excluded, 7951 total
client-release-group-root: Performance for "handle":
client-release-group-root:      fluid-case: 3074ms
client-release-group-root:      dockerfile-copyright-file-header: 1ms
client-release-group-root:      js-ts-copyright-file-header: 128ms
client-release-group-root:      npm-package-metadata-and-sorting: 547ms
client-release-group-root:      npm-private-packages: 501ms
client-release-group-root:      npm-package-license: 474ms
client-release-group-root:      npm-package-json-prettier: 464ms
client-release-group-root:      npm-package-json-script-clean: 457ms
client-release-group-root:      npm-package-json-script-dep: 437ms
client-release-group-root:      npm-package-json-scripts-args: 436ms
client-release-group-root:      npm-package-json-test-scripts: 395ms
client-release-group-root:      npm-package-json-script-mocha-config: 346ms
client-release-group-root:      npm-package-json-script-jest-config: 386ms
client-release-group-root:      npm-package-json-esm: 348ms
client-release-group-root:      npm-package-json-clean-script: 342ms
client-release-group-root:      npm-package-types-field: 331ms
client-release-group-root:      npm-package-exports-field: 136ms
client-release-group-root:      fluid-build-tasks-tsc: 240ms
client-release-group-root:      fluid-build-tasks-eslint: 156ms
client-release-group-root:      pnpm-npm-package-json-preinstall: 1ms
client-release-group-root:      pnpm-lock-workspace: 0ms
client-release-group-root:      pnpm-lock-no-package-lock: 0ms
client-release-group-root:      assert-short-codes: 2633ms
client-release-group-root:      html-copyright-file-header: 2ms
client-release-group-root:      dockerfile-packages: 18ms
client-release-group-root:      package-lockfiles-no-private-url: 1ms
client-release-group-root: Performance for "final":
client-release-group-root:      assert-short-codes: 17ms
[13/13] x client-release-group-root: flub check policy - 5.104s
~~~

Comment (alexvy86) [reply packages/framework/client-logger/app-insights-logger/package.json:78]: Ok, I think instead of adding a `"build:test"` script that does the exact same thing as the `"tsc"` one, we can do [what we did in the common-definitions package](https://github.com/microsoft/FluidFramework/blob/25406b773364cd47e0d57d31db16bc5e94a9cd7d/common/lib/common-definitions/package.json#L75-L80), to explicitly declare that the `tsc` task depends on the `typetests:gen` one. The problem that the policy is complaining about is that if `build:test` doesn't exist but `typetests:gen` does, then we need to make sure that the build of the source code depends on generating the typests, i.e. `typetests:gen` (when `build:test` exists, [our policy already ensures _that one_ depends on `typetests:gen`](https://github.com/microsoft/FluidFramework/blob/25406b773364cd47e0d57d31db16bc5e94a9cd7d/fluidBuild.config.cjs#L54-L58)).

Comment (chentong7) [reply packages/framework/client-logger/app-insights-logger/package.json:78]: I see! That's what it means!! Updating...Thx, Alejandro

**Review by alexvy86 — APPROVED** [review]
